### PR TITLE
Added TFeat, AffNet, OriNet to kornia.features

### PR DESCRIPTION
**Pull request by ducha-aiki**
### Description
Added:

- TFeat from https://github.com/vbalnt/tfeat/, BMVC 2016
- OriNet, AffNet from https://github.com/ducha-aiki/affnet/, ECCV 2018

Also fixed the incorrect doc  description of the current OrientationEstimator
### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x ] Did you discuss the functionality or any breaking changes before ?
- [x ] **Pass all tests**: did you test in local ? `make test`
- [x ] Unittests: did you add tests for your new functionality ?
- [x ] Documentations: did you build documentation ? `make build-docs`
- [x ] Implementation: is your code well commented and follow conventions ? `make lint`
- [x ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [x ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Comment by ducha-aiki** [comment]
I will add jit-stuff to the rest of features in the separate PR to avoid conflicts

File: kornia/feature/affine_shape.py [hunk 1]
```diff
@@ -120,3 +126,93 @@ def forward(self, laf: torch.Tensor, img: torch.Tensor) -> torch.Tensor:  # type
...
+    @staticmethod
+    def _normalize_input(x: torch.Tensor, eps: float = 1e-6) -> torch.Tensor:
+        "Utility function that normalizes the input by batch."""
+        sp, mp = torch.std_mean(x, dim=(-3, -2, -1), keepdim=True)
+        # WARNING: we need to .detach() input, otherwise the gradients produced by
+        # the patches extractor with F.grid_sample are very noisy, making the detector
+        # training totally unstable.
+        return (x - mp.detach()) / (sp.detach() + eps)
+
+    def forward(self, laf: torch.Tensor, img: torch.Tensor) -> torch.Tensor:  # type: ignore
```
Comment (edgarriba) [inline kornia/feature/affine_shape.py:187]: I think type ignore is not needed anymore

File: kornia/feature/affine_shape.py [hunk 2]
```diff
@@ -120,3 +126,93 @@ def forward(self, laf: torch.Tensor, img: torch.Tensor) -> torch.Tensor:  # type
...
+        """
+        Args:
+            laf: (torch.Tensor) shape [BxNx2x3]
+            img: (torch.Tensor) shape [Bx1xHxW]
+
+        Returns:
+            laf_out: (torch.Tensor) shape [BxNx2x3]"""
```
Comment (edgarriba) [inline kornia/feature/affine_shape.py:194]: this won't render well I believe. Try `torch.Tensor: laf_out with shape [BxNx2x3]`

File: kornia/feature/orientation.py [hunk 3]
```diff
@@ -100,28 +103,110 @@ def forward(self, patch: torch.Tensor) -> torch.Tensor:  # type: ignore
...
+    @staticmethod
+    def _normalize_input(x: torch.Tensor, eps: float = 1e-6) -> torch.Tensor:
+        "Utility function that normalizes the input by batch."""
+        sp, mp = torch.std_mean(x, dim=(-3, -2, -1), keepdim=True)
+        # WARNING: we need to .detach() input, otherwise the gradients produced by
+        # the patches extractor with F.grid_sample are very noisy, making the detector
+        # training totally unstable.
+        return (x - mp.detach()) / (sp.detach() + eps)
+
+    def forward(self, patch: torch.Tensor) -> torch.Tensor:  # type: ignore
```
Comment (edgarriba) [inline kornia/feature/orientation.py:173]: no type ignore

File: kornia/feature/orientation.py [hunk 4]
```diff
@@ -100,28 +103,110 @@ def forward(self, patch: torch.Tensor) -> torch.Tensor:  # type: ignore
...
             patch_size: int, default = 32
-            num_angular_bins: int, default is 36"""
+            num_angular_bins: int, default is 36
+            angle_detector: nn.Module. Patch orientation estimator, e.g. PatchDominantGradientOrientation or OriNet. Default: None """ # noqa pylint: disable
 
     def __init__(self,
                  patch_size: int = 32,
-                 num_angular_bins: int = 36):
+                 num_angular_bins: int = 36,
+                 angle_detector: Optional[nn.Module] = None):  # type: ignore
```
Comment (edgarriba) [inline kornia/feature/orientation.py:196]: type ignore

File: kornia/feature/tfeat.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            nn.Tanh()
+        )
+        # use torch.hub to load pretrained model
+        if pretrained:
+            pretrained_dict = torch.hub.load_state_dict_from_url(
+                urls['liberty'], map_location=lambda storage, loc: storage
+            )
+            self.load_state_dict(pretrained_dict, strict=True)
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:   # type: ignore
```
Comment (edgarriba) [inline kornia/feature/tfeat.py:58]: no type ignore

**Review by edgarriba — APPROVED** [review]
LGTM. Just minor things
